internal/services/user_management/user_management.go

```diff
@@ -0,0 +1,115 @@
+package user_management
+
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"Tradeasy/internal/provider/redis"
+	"Tradeasy/internal/utils"
+	"errors"
+	"strings"
+	"time"
+)
+
+var ExpiryTime = 5 * time.Minute
+
+func SignUp(SUpReq SignUpRequest) (SUpRes SignUpResponse, err error) {
+	email := SUpReq.EmailId
+
+	var user model.Users
+	err = config.DB.Table("users").Where("email_id = ?", SUpReq.EmailId).First(&user).Error
+	if err == nil {
+		return SUpRes, errors.New("email id already registered")
+	}
+	err = config.DB.Table("users").Where("password = ?", SUpRes.Password).First(&user).Error
+	if err == nil {
+		return SUpRes, errors.New("password already taken")
+	}
+	user.UserId = SUpRes.UserId
+	user.EmailId = SUpReq.EmailId
+	user.Password = SUpReq.Password
+	err = config.DB.Table("users").Create(&user).Error
+	if err != nil {
+		return SUpRes, errors.New("signUp failed")
+	}
+	SUpRes.UserId = strings.Split(email, "@")[0]
+	SUpRes.Password = SUpReq.Password
+	SUpRes.Message = "User registered"
+	return SUpRes, nil
+}
+
+func UserDetails(detReq UserDetailsRequest, userid string) (detRes UserDetailsResponse, err error) {
+
+	var user model.Users
+	var ta model.TradingAccount
+
+	err = config.DB.Table("users").Where("user_id = ? ", userid).First(&user).Error
+	if err != nil {
+		return detRes, errors.New("user not found")
+	}
+	err = config.DB.Table("trading_account").Where("user_id = ? OR pan_card_no = ? OR bank_acc_no = ?", userid, detReq.PanCardNo, detReq.BankAccNo).First(&ta).Error
+	if err == nil {
+		return detRes, errors.New("user details already registered")
+	}
+	ta.UserId = userid
+	ta.PanCardNo = detReq.PanCardNo
+	ta.BankAccNo = detReq.BankAccNo
+	ta.TradingAccId = detRes.TradingAccId
+	ta.Balance = detRes.Balance
+
+	err = config.DB.Table("trading_account").Create(&ta).Error
+	if err != nil {
+		return detRes, errors.New("user details failed to enter")
+	}
+	detRes.TradingAccId = "TRA" + userid
+	detRes.Balance = 0
+	detRes.Message = "User Details registered"
+	return detRes, nil
+}
+func UserSignIn(SInReq SignInRequest) (SInRes SignInResponse, err error) {
+
+	var user model.Users
+	err = config.DB.Table("users").Where("user_id = ?", SInReq.UserId).First(&user).Error
+	if err != nil {
+		return SInRes, errors.New("user not found")
+	}
+	err = config.DB.Table("users").Where("password = ?", SInReq.Password).First(&user).Error
```

Comment:
why are we just checking password here? what about user_id?

Reply:
I am checking user_id also above password

Reply:
Add user_id. Incorrect login otherwise.